www-src/app/replicator/init.coffee

```diff
@@ -0,0 +1,434 @@
+compareVersions = require('../lib/compare_versions').compareVersions
+
+log = require('../lib/persistent_log')
+    date: true
+    processusTag: "Init"
+
+module.exports = class Init
+
+    _.extend Init.prototype, Backbone.Events
+    _.extend Init.prototype, Backbone.StateMachine
+
+    # Override this function to use it as initialize.
+    startStateMachine: ->
+        @initialize()
+        Backbone.StateMachine.startStateMachine.bind(@)(arguments)
+
+    initialize: ->
+        @migrationStates = {}
+
+        @listenTo @, 'transition', (leaveState, enterState) ->
+            log.info "Transition from state #{leaveState} \
+                      to state #{enterState}"
+
+
+    initMigration: ->
+        oldVersion = app.replicator.config.get 'appVersion'
+        for version, migration of @migrations
+            if compareVersions(version, oldVersion) <= 0
+                break
```

Comment:
Im' not really sure how this function works. But shouldn't be it next instead of break?